src/terminal/adapter/FontBuffer.cpp

```diff
@@ -0,0 +1,603 @@
+            return { 15, 12, textWidth };
+        }
+        else if (inRange(10, 16))
+        {
+            // VT420 & VT5xx - 10x16 cell with default text width.
+            return { 10, 16, textWidth };
+        }
+        else if (inRange(10, 20))
+        {
+            // VT340 - 10x20 cell with default text width.
+            return { 10, 20, textWidth };
+        }
+        else if (inRange(12, 30))
+        {
+            // VT382 - 12x30 cell with default text width.
+            return { 12, 30, textWidth };
+        }
+        else
+        {
+            // If all else fails, assume the maximum size.
+            return { MAX_WIDTH, MAX_HEIGHT, textWidth };
+        }
+    }
+    else
+    {
+        if (inRange(6, 10) && noDeclaredSize)
+        {
+            // VT240 - 6x10 cell with no clipping.
+            return { 6, 10, 0 };
+        }
+        else if (inRange(9, 12))
+        {
+            // VT320 - 9x12 cell with default text width.
+            return { 9, 12, textWidth };
+        }
+        else if (inRange(6, 16))
+        {
+            // VT420 & VT5xx - 6x16 cell with default text width.
+            return { 6, 16, textWidth };
+        }
+        else if (inRange(6, 20))
+        {
+            // VT340 - 6x20 cell with default text width.
+            return { 6, 20, textWidth };
+        }
+        else if (inRange(7, 30))
+        {
+            // VT382 - 7x30 cell with default text width.
+            return { 7, 30, textWidth };
+        }
+        else
+        {
+            // If all else fails, assume the maximum size.
+            return { MAX_WIDTH, MAX_HEIGHT, textWidth };
+        }
+    }
+}
+
+void FontBuffer::_packAndCenterBitPatterns()
+{
+    // If this is a text font, we'll clip the bits up to the text width and
+    // center them within the full cell width. For a full cell font we'll just
+    // use all of the bits, and no offset will be required.
+    _textWidth = _textWidth ? _textWidth : _fullWidth;
+    _textWidth = std::min(_textWidth, _fullWidth);
+    _textOffset = (_fullWidth - _textWidth) / 2;
+    const auto textClippingMask = ~(0xFFFF >> _textWidth);
+
+    // If the text is given an explicit width, we check to what extent the
+    // content is offset from center. Knowing that information will enable the
+    // renderer to scale the font more symmetrically.
+    _textCenteringHint = _declaredWidth ? _fullWidth - (_declaredWidth + _textOffset * 2) : 0;
+
+    // Initially the characters are written to the buffer assuming the maximum
+    // cell height, but now that we know the true height, we need to pack the
+    // buffer data so that each character occupies the exact number of scanlines
+    // that are required.
+    for (auto srcLine = 0u, dstLine = 0u; srcLine < _buffer.size(); srcLine++)
+    {
+        if ((srcLine % MAX_HEIGHT) < _fullHeight)
+        {
+            auto characterScanline = _buffer.at(srcLine);

```

Comment:
FYI: Since we know that `srcLine` is always smaller than `_buffer.size()`, you can use `til::at(_buffer, srcLine)` and `til::at(_buffer, dstLine)`.

Reply:
Good point. I've updated those now.